src/Raven.Server/Documents/Archive/ArchivalStorage.cs

```diff
@@ -0,0 +1,356 @@
+                    }
+                    
+                    foreach (var counterNameKey in counterNamesDict.GetPropertyNames())
+                    {
+                        builder.WritePropertyName(counterNameKey);
+                        builder.WriteValue((LazyStringValue)counterNamesDict[counterNameKey]);
+                    }
+                }
+                
+                builder.WriteObjectEnd();
+                
+                builder.WriteObjectEnd();
+                builder.FinalizeDocument();
+
+                mergedCounterGroupValues = builder.CreateReader();
+            }
+
+            return mergedCounterGroupValues;
+        }
+
+        
+        private void ArchiveDocumentCountersIfNeeded(DocumentsOperationContext context, Document doc, string documentId, string collectionName, string archivedCollectionName)
+        {
+            if ((doc.Flags & DocumentFlags.HasCounters) == 0) return;
+            
+            var counterGroups = _database.DocumentsStorage.CountersStorage.GetCounterValuesForDocument(context, documentId).ToList();
+                                            
+            var clonedCountersGroupsValues = new List<BlittableJsonReaderObject>(counterGroups.Count);
+            foreach (var counterGroup in counterGroups)
+                clonedCountersGroupsValues.Add(counterGroup.Values.Clone(context));
+                                            
+            var mergedCounterGroupsValues = BuildMergedCounterGroupValues(context, clonedCountersGroupsValues, documentId);
+                                                
+            _database.DocumentsStorage.CountersStorage.DeleteCountersForDocument(context, documentId, new CollectionName(collectionName));
+                                            
+            var newCounterEtag = _database.DocumentsStorage.GenerateNextEtag();
+            var newCounterChangeVector = _database.DocumentsStorage.GetNewChangeVector(context, newCounterEtag);
+            _database.DocumentsStorage.CountersStorage.PutCounters(context, documentId, archivedCollectionName, newCounterChangeVector , mergedCounterGroupsValues, newCounterEtag);
+        }
+
+        private void ArchiveDocumentTimeSeriesIfNeeded(DocumentsOperationContext context, Document doc, string documentId, string collectionName, string archivedCollectionName)
+        {
+            
+        }
+        
+        private void ArchiveDocumentRevisionsIfNeeded(DocumentsOperationContext context, Document doc, string documentId, string collectionName, string archivedCollectionName)
+        {
+            
+        }
+        
+        private void ArchiveDocumentAttachmentsIfNeeded(DocumentsOperationContext context, Document doc, string documentId, string collectionName, string archivedCollectionName)
+        {
+                    
+        }
+
+        private bool MigrateDocument(DocumentsOperationContext context, (Slice LowerId, string Id) ids, DateTime currentTime)
```

Comment:
Let's avoid tuples in args

Reply:
Fixed in https://github.com/ravendb/ravendb/pull/15715/commits/1b8a0885aa25f24665c0a2a8e219ff621cfa2808